src/System.Private.CoreLib/shared/System/Diagnostics/StackTrace.cs

```diff
@@ -204,8 +205,8 @@ internal string ToString(TraceFormat traceFormat)
             StringBuilder sb = new StringBuilder(255);
             for (int iFrameIndex = 0; iFrameIndex < _numOfFrames; iFrameIndex++)
             {
-                StackFrame sf = GetFrame(iFrameIndex);
-                MethodBase mb = sf.GetMethod();
+                StackFrame? sf = GetFrame(iFrameIndex);
+                MethodBase? mb = sf?.GetMethod();
```

Comment:
This actually changes the IL, from `sf.GetMethod()` to `sf != null ? sf.GetMethod() : null`, which we should avoid unless it's actually a problem (which I doubt it is).  In what situation can `sf` actually be `null` here?

Reply:
@jaredpar, per our conversation the other day, an example of where these sneak in.

Reply:
Though interestingly, this is a case where technically the compiler is right, as this (super uncommon use) would produce a NullReferenceException:
```C#
new StackTrace((StackFrame?)null).ToString();
```
Given all of the work that's already being done here to create this stack trace, adding another null check shouldn't hurt.  Ok.

cc: @jkotas

Reply:
Actually I just noticed that we had a test in corefx expecting a NRE... Maybe that was intentional?

Or we should just disable that test... we're hitting it in the merge into master from the NullableFeature.
https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_ubuntu_corefx_innerloop_prtest/10683/testReport/junit/System.Diagnostics.Tests/StackTraceTests/ToString_NullFrame_ThrowsNullReferenceException/